File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        result=set()

        for i in range(len(nums)):
            result.add(nums[i])

        total=0
        m=len(result)

        for i in range(len(nums)):
            count=1

            if nums[i]+1 in result and nums[i]-1 not in result:
                result.remove(nums[i])
                cur=nums[i]+1

                while cur in result:
                    result.remove(cur)
                    cur=cur+1
                    count+=1

            total=max(total,count)

        return total if m>1 else m

                    




        

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set=new HashSet<>();
        for(int num:nums){
            set.add(num);
        }


        int total=0;
        int n=nums.length;
        for(int i=0;i<n;i++){
            int count=1;
            if(set.contains(nums[i]+1) && !set.contains(nums[i]-1)){
                set.remove(nums[i]);
                int cur=nums[i]+1;
                while(set.contains(cur)){
                    set.remove(cur);
                    cur++;
                    count++;
                }

            }
            total=Math.max(total,count);
        }
        return total;

    }
}